preprocessing/shift_generation.py

```diff
@@ -533,7 +534,7 @@ def get_durations(time_step):
 
     t_weekly_off = WEEKLY_REST_DURATION[0]
 
-    while t_weekly_off <= WEEKLY_REST_DURATION[1]:
+    while t_weekly_off <= min(WEEKLY_REST_DURATION[1], MAX_REWARDED_WEEKLY_REST):
```

Comment:
You are sure that this should not be `max`?  As it stands now it seems to have no effect.

Reply:
Yes. Weekly_rest_duration[1] is now 90. We are not rewarding longer off shifts than 72, so the implicit do not have to lokk for longer off-periods than that.

Reply:
I thought this was generating shifts for the given duration, and this should actually create shifts of length 90. If I am mistaken I will approve right away

Reply:
The implicit model will look for `min(WEEKLY_REST_DURATION[1], MAX_REWARDED_WEEKLY_REST)` hours where an employee is not working. If it can find 90 hours it will find 72 hours! But there is no point in finding 90 hours if only 72 is rewarded. 

In the other models we are creating explicit off-shifts, while we in implicit model only look for connected time intervasls.